libvips/arithmetic/math.c

```diff
@@ -129,6 +129,28 @@ vips_math_build( VipsObject *object )
 		g_assert_not_reached(); \
 	} 
 
+/* inverse hyperbolic functions for old compilers
+   they were added to the standard in C99 and C++11
+*/
+#if ( \
+  (defined(__cplusplus) && __cplusplus >= 201103L) \
+  || (defined(__STDC__) && __STDC_VERSION__ >= 199901L) \
+)
+#define HAS_INVERSE_HYPERBOLICS 1
+#else
+#define HAS_INVERSE_HYPERBOLICS 0
+#endif
+
+#if HAS_INVERSE_HYPERBOLICS
+  #define ASINH sinh
+  #define ACOSH cosh
+  #define ATANH tanh
+#else
+  #define ASINH( X ) log((X) + sqrt( (X)*(X)+1 ))
+  #define ACOSH( X ) log((X) + sqrt( (X)*(X)-1 ))
+  #define ATANH( X ) 0.5 * log( (1+(X)) / (1-(X)) )
```

Comment:
I'd put an extra `()` around the whole expression to stop precedence messing this up if used in more complex expressions.